File: frameworks/Java/fair/java-kora1-jdbc-repository/src/main/java/io/techempower/benchmark/kora/controller/JteModule.java
package io.techempower.benchmark.kora.controller;

import io.techempower.benchmark.kora.model.Fortune;
import io.techempower.benchmark.kora.util.JteUtils;
import ru.tinkoff.kora.common.Module;
import ru.tinkoff.kora.http.common.body.HttpBody;
import ru.tinkoff.kora.http.server.common.HttpServerResponse;
import ru.tinkoff.kora.http.server.common.handler.HttpServerResponseMapper;

import java.util.List;

@Module
public interface JteModule {

    String CONTENT_TYPE = "text/html;charset=UTF-8";

    default HttpServerResponseMapper<List<Fortune>> jteListHttpServerResponseMapper() {
        return (ctx, request, result) -> HttpServerResponse.of(200, request.headers(), HttpBody.of(CONTENT_TYPE, JteUtils.serializeStandard(result)));
    }
}

File: frameworks/Java/fair/kotlin-kora1-vertx-repository-suspend/src/main/kotlin/io/techempower/benchmark/kora/controller/JteModule.kt
package io.techempower.benchmark.kora.controller

import io.techempower.benchmark.kora.model.Fortune
import io.techempower.benchmark.kora.util.JteUtils
import ru.tinkoff.kora.common.Module
import ru.tinkoff.kora.http.common.body.HttpBody
import ru.tinkoff.kora.http.server.common.HttpServerResponse
import ru.tinkoff.kora.http.server.common.handler.HttpServerResponseMapper

@Module
interface JteModule {

    fun jteListHttpServerResponseMapper(): HttpServerResponseMapper<List<Fortune>> {
        return HttpServerResponseMapper { _, request, result ->
            HttpServerResponse.of(200, request.headers(), HttpBody.of("text/html;charset=UTF-8", JteUtils.serializeStandard(result)))
        }
    }
}
